basics_of_sst-core/tutorial/Example01.md

```diff
@@ -0,0 +1,436 @@
+# Communicating Between Two Components
+
+Communication between components in SST is accomplished by passing events over links that connect the components.  This example will demonstrate how to send messages between two components in SST.  To show that the components are really communicating the component clock event handler will be modified to increment the clock tick count when a message is received and end the simulation when the specified number of clock counts have been received.
+
+For this example we'll need to make the following changes.
+
+1.  Add a link to the component over which the message will be sent.
+2.  Define the message to be sent.
+3.  Update the clock event handler to process the incoming messages.
+4.  Configure the simulation to connect two components via the link.
+
+
+## Example Setup
+
+To set up for this example, perform the following steps:
+
+1.  Navigate to the `example` folder created in `Example00` and create a folder named `Example01`.
+2.  Copy the files created as part of `Example00` \(`ExampleComponent.cc`, `ExampleComponent.h`, `ExampleConfig.py`, and `Makefile`\) to the `Example01` folder.
+3.  Open the copied `ExampleComponent.h` file and change the namespace to `Example01`.
+4.  Open the copied `ExampleComponent.cc` file and change the `using namespace` directive to refer to the `Example01` namespace.
+
+## Adding the Link
+
+An SST link is an object within an SST component over which messages are sent.  Its name is a bit of a misnomer since a link is normally thought of as the actual connection between two objects, like the telephone line between two houses or the network cable between two computers.  In fact, the SST code interface and documentation seems to use the names "link" and "port" interchangeably.  For this example, the term "link" will be used for consistency unless the SST interface specifically uses the term "port".
```

Comment:
"link" and "port" are not interchangeable. In the telephone analogy, "link" would be the connection between two phone numbers, while "port" would be the telephone numbers. The link is the actual connection while the port is where a link attaches to a component.